detection_rules/integrations.py

```diff
@@ -89,6 +162,49 @@ def find_least_compatible_version(package: str, integration: str,
     raise ValueError(f"no compatible version for integration {package}:{integration}")
 
 
+def find_compatible_version_window(package: str, integration: str,
+                                   rule_stack_version: str, packages_manifest: dict) -> Generator[str, None, None]:
+    """Finds least compatible version for specified integration based on stack version supplied."""
+
+    if not package:
+        raise ValueError("Package must be specified")
+
+    package_manifest = packages_manifest.get(package)
+    if package_manifest is None:
+        raise ValueError(f"Package {package} not found in manifest.")
+
+    integration_manifests = sorted(package_manifest.items(), key=lambda x: Version(str(x[0])))
+
+    # iterates through ascending integration manifests
+    # returns latest major version that is least compatible
+    for version, manifest in integration_manifests:
+        kibana_conditions = manifest.get("conditions", {}).get("kibana", {})
+        version_requirement = kibana_conditions.get("version")
+        if not version_requirement:
+            raise ValueError(f"Manifest for {package}:{integration} version {version} is missing conditions.")
+
+        compatible_versions = re.sub(r"\>|\<|\=|\^", "", manifest["conditions"]["kibana"]["version"]).split(" || ")
+
+        if not compatible_versions:
+            raise ValueError(f"Manifest for {package}:{integration} version {version} is missing compatible versions")
+
+        if len(compatible_versions) > 1:
+            highest_compatible_version = max(compatible_versions, key=lambda x: Version(x))
+
+            if Version(highest_compatible_version) > Version(rule_stack_version):
+                print(f"Integration {package}-{integration} {version=} has multiple stack version requirements.",
+                      f"Consider updating min_stack version to the latest {compatible_versions}.")
```

Comment:
Design Q: Do we break here and force the rule developer to update the min_stack or just warn?